server/procedures/__tests__/verifyUser.test.ts

```diff
@@ -0,0 +1,85 @@
+import User from '../../database/models/user';
+import { clearDB } from '../../database/queries';
+import seedDB from '../../database/seeders';
+import { generateJWT } from '../../helpers';
+import { mockUser } from '../mockData';
+import verifyUser, { alreadyResponse, successResponse } from '../verifyUser';
+
+describe('Verify User rpc', () => {
+  const clientCallback = jest.fn();
+  const user = new User(mockUser);
+
+  beforeAll(async (done) => {
+    await clearDB();
+    await seedDB();
+    done();
+  });
+
+  afterAll(async (done) => {
+    await clearDB();
+    done();
+  });
+
+  afterEach(() => {
+    clientCallback.mockReset();
+  });
+
+  it('should throw error if the token was signed using wrong narrative', async (done) => {
+    const emailToken = generateJWT(user, undefined, 'wrong narrative');
+    // @ts-ignore
+    await verifyUser({ request: { token: emailToken } }, clientCallback);
+    const [[arg1]] = clientCallback.mock.calls;
+    expect(arg1.message).toEqual(
+      'Expected the payload\'s narrative to equal verification but received wrong narrative!',
+    );
+    done();
+  });
+
+  it('should throw error if the user is not found', async (done) => {
+    const emailToken = generateJWT(user, undefined, 'verification');
+    // @ts-ignore
+    await verifyUser({ request: { token: emailToken } }, clientCallback);
+    const [[arg1]] = clientCallback.mock.calls;
+    expect(arg1.message).toEqual(
+      'User not found!',
+    );
+    done();
+  });
+
+  it('should verify the user and update their details in the db', async (done) => {
+    await user.save();
+    const emailToken = generateJWT(user, undefined, 'verification');
+    // @ts-ignore
+    await verifyUser({ request: { token: emailToken } }, clientCallback);
+    const storedUser = await User.findUserByEmail(mockUser.email);
+    if (!storedUser) {
+      expect(1).toEqual(2);
```

Comment:
What does this do?

Reply:
TS knows that we return false if the user is not found. So when we try assertions on some user attributes later in the test, TS complains about it, unless we suppress each line with `//@ts-ingore`, which in my opinion, doesn't look as great.

So, if false is returned, it means that the user is not found and that the tests should fail there. The return statement after the `expect (1)toEqual.(2)` let's TS know that this function will exit of false is returned and if we continue execution, then it will be on type `User` and it won't complain anymore, hence removing the `//@ts-ignore` lines, making code look cleaner.